tornjak-frontend/src/components/entry-create.tsx

```diff
@@ -132,122 +203,172 @@ class CreateEntry extends Component {
         this.prepareSelectorsList();
       }
 
-      if (prevState.parentId !== this.state.parentId)  {
-          this.prepareSelectorsList();
+      if (prevState.parentId !== this.state.parentId) {
+        this.prepareSelectorsList();
       }
     }
   }
 
-  prepareParentIdAgentsList() {
+  prepareParentIdAgentsList(): void {
     var idx = 0, prefix = "spiffe://";
-    let localAgentsIdList = [];
-    if (this.props.globalServerInfo.length === 0) {
-      return
-    }
+    let localAgentsIdList: string[] = [];
+    if (Object.keys(this.props.globalServerInfo).length === 0) { return }
     //user prefered option
     localAgentsIdList[0] = this.state.parentIdManualEntryOption;
     //default option
     localAgentsIdList[1] = prefix + this.props.globalServerInfo.trustDomain + "/spire/server";
 
     //agents
-    let agentEntriesDict = this.SpiffeHelper.getAgentsEntries(this.props.globalAgentsList, this.props.globalEntriesList)
+    // note: any for now - will be specifically typed when spiffehelper file is typed - gives error now since its not typed
+    let agentEntriesDict: any = this.SpiffeHelper.getAgentsEntries(this.props.globalAgentsList, this.props.globalEntriesList) //remove any after adding types to helpers
```

Comment:
if this is not done, let's add a TODO(mamy-CS) flag so we can scan through at the end to verify all of them are handled

Reply:
Good point, Done!